stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelNameFormatter.kt

```diff
@@ -0,0 +1,68 @@
+package io.getstream.chat.android.compose.ui.util
+
+import android.content.Context
+import io.getstream.chat.android.client.extensions.getUsersExcludingCurrent
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.compose.R
+
+/**
+ *  An interface that allows to generate a name for the given channel.
+ */
+public interface ChannelNameFormatter {
+
+    /**
+     * Generates a name for the given channel.
+     *
+     * @param channel The channel whose data is used to generate the name.
+     * @return The display name for the given channel.
+     */
+    public fun format(channel: Channel): String
+
+    public companion object {
+        /**
+         * Build the default channel name formatter.
+         *
+         * @param context The context to load string resources.
+         * @param maxMembers The maximum number of members used to generate a name for a distinct channel.
+         *
+         * @see [DefaultChannelNameFormatter]
+         */
+        public fun defaultFormatter(
+            context: Context,
+            maxMembers: Int = 5,
+        ): ChannelNameFormatter {
+            return DefaultChannelNameFormatter(context, maxMembers)
+        }
+    }
+}
+
+/**
+ * A simple implementation of [ChannelNameFormatter] that allows to generate a name for a channel
+ * based on the following rules:
+ *
+ * - If the channel has a name, then its name is returned
+ * - If the channel is distinct, then a comma-separated list of member names is returned
+ * - Otherwise, the placeholder text defined in [R.string.stream_compose_untitled_channel] is returned
+ *
+ * @param context The context to load string resources.
+ * @param maxMembers The maximum number of members used to generate a name for a distinct channel.
+ */
+private class DefaultChannelNameFormatter(
+    private val context: Context,
+    private val maxMembers: Int,
+) : ChannelNameFormatter {
```

Comment:
I'm not sure but how about adding the `@Composable` annotation to the `format` method and removing the `context `argument? We may get the context from the `LocalContext.current` instead.

```kotlin
@Composable
override fun format(channel: Channel): String {
    val context: Context = LocalContext.current
    return channel.name.takeIf { it.isNotEmpty() }
        ?: channel.getUsersExcludingCurrent()
            .joinToString(limit = maxMembers) { it.name }
            .takeIf { it.isNotEmpty() }
        ?: context.getString(R.string.stream_compose_untitled_channel)
}
```

Reply:
That's actually a very good question. I actually don't have a strong opinion about that.

Reply:
@skydoves @bychkovdmitry Really nice discussion.

Generally, adding @Composable to functions that return values is not that common and should serve more less only for functions that really do depend on the scope and context of the Compose tree.

E.g. if you're observing state that changes and the function needs to recompose. Or if you're, based on the compose state, lauching extra lifecycle things that affect the Compose tree in some way, like system back click handlers updating the navigation or the active screen.

In this case, although we depend on the Context, nothing mutates within the Compose tree to affect our state or result, as this result will always be the same, regardless of the context object (it's just used to fetch the resource).

So in that sense, I prefer it this way and I think it's good, maybe we could split it up into a smaller function that might be more descriptive, but it's also fine this way!